rpc/types/src/json.rs

```diff
@@ -92,6 +169,23 @@ define_request_and_response! {
     // "Flatten" means the field(s) of a struct gets inlined
     // directly into the struct during (de)serialization, see:
     // <https://serde.rs/field-attrs.html#flatten>.
+    #[doc = serde_doc_test!(
+        GET_BLOCK_TEMPLATE_RESPONSE => GetBlockTemplateResponse {
+            base: ResponseBase::ok(),
+            blockhashing_blob: "1010f4bae0b4069d648e741d85ca0e7acb4501f051b27e9b107d3cd7a3f03aa7f776089117c81a00000000e0c20372be23d356347091025c5b5e8f2abf83ab618378565cce2b703491523401".into(),
+            blocktemplate_blob: "1010f4bae0b4069d648e741d85ca0e7acb4501f051b27e9b107d3cd7a3f03aa7f776089117c81a0000000002c681c30101ff8a81c3010180e0a596bb11033b7eedf47baf878f3490cb20b696079c34bd017fe59b0d070e74d73ffabc4bb0e05f011decb630f3148d0163b3bd39690dde4078e4cfb69fecf020d6278a27bad10c58023c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000".into(),
+            difficulty_top64: 0,
+            difficulty: 283305047039,
+            expected_reward: 600000000000,
+            height: 3195018,
+            next_seed_hash: "".into(),
+            prev_hash: "9d648e741d85ca0e7acb4501f051b27e9b107d3cd7a3f03aa7f776089117c81a".into(),
+            reserved_offset: 131,
+            seed_hash: "e2aa0b7b55042cd48b02e395d78fa66a29815ccc1584e38db2d1f0e8485cd44f".into(),
+            seed_height: 3194880,
+            wide_difficulty: "0x41f64bf3ff".into(),
+        }
+    )]
     ResponseBase {
```

Comment:
Used on responses as well.

I had second thoughts about putting the tests directly on-top and in the same file as the types themselves since it's a bit noisy and makes the files quite long but:

1. There's only 1 doc-test per type
2. The test is as close as possible to the definition site; better for maintenance

I think I'd be okay with a separate file for tests as well, downside is that it separates types and tests and it won't be a doc-test, just a normal test.

Reply:
I think this is ok, although the type definitions are starting to get pretty confusing...